sbi/inference/posterior.py

```diff
@@ -129,12 +122,89 @@ def set_default_x(self, x: Tensor) -> "NeuralPosterior":
         Returns:
             `NeuralPosterior` that will use a default `x` when not explicitly passed.
         """
-        processed_x = process_x(x, self._x_shape)
-        self._warn_if_posterior_was_focused_on_different_x(processed_x)
-        self._x = processed_x
 
         return self
 
+    @property
+    def mcmc_method(self) -> str:
+        """Returns MCMC method."""
+        return self._mcmc_method
+
+    @mcmc_method.setter
+    def mcmc_method(self, method: str) -> None:
+        """See `set_mcmc_method`."""
+        self.set_mcmc_method(method)
+
+    def set_mcmc_method(self, method: str) -> "NeuralPosterior":
+        """Set sampling method to for MCMC.
+
+        Args:
+            method: Method to use.
+
+        Returns:
+            `NeuralPosterior` for chainable calls.
+        """
+        self._mcmc_method = method
+        return self
+
+    @property
+    def mcmc_parameters(self) -> dict:
+        """Returns MCMC parameters."""
+        if self._mcmc_parameters is None:
+            return {}
+        else:
+            return self._mcmc_parameters
+
+    @mcmc_parameters.setter
+    def mcmc_parameters(self, parameters: Dict[str, Any]) -> None:
+        """See `set_mcmc_parameters`."""
+        self.set_mcmc_parameters(parameters)
+
+    def set_mcmc_parameters(self, parameters: Dict[str, Any]) -> "NeuralPosterior":
+        """Set parameters for MCMC.
+
+        Args:
+            parameters: MCMC parameters to use.
+
+        Returns:
+            `NeuralPosterior` for chainable calls.
+        """
+        self._mcmc_parameters = parameters
+        return self
+
+    @property
+    def sample_with_mcmc(self) -> bool:
+        """Return `True` if NeuralPosterior instance is configured to use MCMC in `.sample()`."""
+        return self._sample_with_mcmc
+
+    @sample_with_mcmc.setter
+    def sample_with_mcmc(self, value: bool) -> None:
+        """See `set_sample_with_mcmc`."""
+        self.set_sample_with_mcmc(value)
+
+    def set_sample_with_mcmc(self, value: bool) -> "NeuralPosterior":
```

Comment:
Suggest `on` instead of `value`. `on=True` seems more readable to me. Another option: `use_mcmc`